File: JavaScript_Courses/JavaScript_Basics/Preparation/06_Nested_Loops_Exercise/02_Equal_Sums_Even_Odd_Position.js
function equalSumsEvenOddPosition(input) {
    let intervalStart = Number(input[0]);
    let intervalEnd = Number(input[1]);

    let result = '';

    for (let num = intervalStart; num <= intervalEnd; num++) {
        let evenSum = 0;
        let oddSum = 0;

        let numAsString = String(num);
        // for (let i = 0; i < 6; i++) {
        for (let i = 0; i < numAsString.length; i++) {
            if (i % 2 == 0) {
                evenSum += Number(numAsString[i]);
            } else {
                oddSum += Number(numAsString[i]);
            }
        }

        if (evenSum == oddSum) {
            result += `${numAsString} `;
        }
    }

    console.log(result);
}


// equalSumsEvenOddPosition(["100000", "100050"]);

File: Python_Courses/Python_Basics/Python_PB_2023_10/06_Nested_Loops_Exercise/02_Equal_Sums_Even_Odd_Position.py
first_num = int(input())
second_num = int(input())

for num in range(first_num, second_num + 1):
    num_as_string = str(num)
    even_sum = 0
    odd_sum = 0
    for idx, digit in enumerate(num_as_string):
        if idx % 2 == 0:
            even_sum += int(digit)
        else:
            odd_sum += int(digit)

    if even_sum == odd_sum:
        print(num, end=" ")
